pkg/sql/schemachanger/scplan/plan.go

```diff
@@ -108,14 +108,35 @@ func MakePlan(ctx context.Context, initial scpb.CurrentState, params Params) (p
 func makePlan(ctx context.Context, p *Plan) (err error) {
 	{
 		start := timeutil.Now()
+		var oldCurrent []scpb.Status
+		if p.Params.ExecutionPhase <= scop.PreCommitPhase {
```

Comment:
I find this resetting confusing for statement phase. Are we resetting also at each statement phase? The comments say that we reset at pre-commit time. Similar language is used in the stage builder.

Reply:
We're _not_ resetting the state at statement phase. It appears that way here only because we want to generate the *full graph*, once that's done we set it back to what it was. Why don't we deep-copy the state and reset the copy you might ask: that's because there's a bunch of code which relies on pointer equalities.

Reply:
Nothing depends on status by pointer equality, does it?

Reply:
Not that I can think of. I swap the current state out in a less bone-headed way in my latest revision, I hope it addresses your concerns.